pkg/traceroute/trace.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package traceroute
+
+import "net"
+
+type Trace struct {
```

Comment:
I am thinking this might be better as an interface, given all the 4 and 6 variation I see everywhere. 
Take a look at net.Conn.
WDYT?

Reply:
`net.Conn` or `net.PacketConn` does not allow the manipilation of Time-To-Live (TTL) or HopLimit for the packets, which is essential to traceroute. So I have to go with `x/net/ipv4.PacketConn` and `x/net/ipv6.PacketConn` to get it done. These types do no implemenet the `net.Conn` interface of the standard library.

I prolly have to wrap this in an interface myself to make it address family independant.

Reply:
Turns out: I cant even wrap `x/net/ipv4` and `x/net/ipv6` in an interface. Too many distinct functions, structs and fieldnames. It would require to check the type and cast the interface into the specific type all the time. This leads the idea of implementing the interface ad absurdum.